src/main/java/moze_intel/projecte/network/commands/EMCCMD.java

```diff
@@ -0,0 +1,137 @@
+				.then(Commands.argument("player", EntityArgument.player())
+					.executes((ctx) -> handle(ctx, ActionType.GET))
+				)
+			);
+	}
+
+	private static ArgumentBuilder<CommandSourceStack, ?> executeWithParameters(ActionType actionType) {
+		return Commands.argument("player", EntityArgument.player())
+			.then(Commands.argument("value", StringArgumentType.string())
+				.executes(ctx -> handle(ctx, actionType))
+			);
+	}
+	private static MutableComponent formatEMC(BigInteger emc) {
+		return TextComponentUtil.build(ChatFormatting.GRAY, TransmutationEMCFormatter.formatEMC(emc));
+	}
+
+	private static int handle(CommandContext<CommandSourceStack> ctx, ActionType action) throws CommandSyntaxException {
+		ServerPlayer player = EntityArgument.getPlayer(ctx, "player");
+		Optional<IKnowledgeProvider> cap = player.getCapability(PECapabilities.KNOWLEDGE_CAPABILITY).resolve();
+		if (cap.isEmpty()) {
+			ctx.getSource().sendFailure(PELang.COMMAND_PROVIDER_FAIL.translateColored(ChatFormatting.RED));
+			return 0;
+		}
+		IKnowledgeProvider provider = cap.get();
+		if (action == ActionType.GET) {
+			ctx.getSource().sendSuccess(PELang.COMMAND_EMC_GET_SUCCESS.translate(player.getDisplayName(), formatEMC(provider.getEmc())), true);
+			return 1;
+		}
+		String val = StringArgumentType.getString(ctx, "value");
+		@Nullable BigInteger value = null;
+
+		try {
+			value = new BigInteger(val);
+			if (value.compareTo(BigInteger.ZERO) < 0) {
+				switch (action) {
+					case ADD, REMOVE -> {
+						action = action == ActionType.ADD ? ActionType.REMOVE : ActionType.ADD;
+						value = value.abs();
+					}
+					case SET, TEST -> {
+						value = null;
+					}
+				}
+			}
+		} catch (NumberFormatException ignore) {
+		}
+		if(value == null) {
+			ctx.getSource().sendFailure(PELang.COMMAND_EMC_INVALID.translateColored(ChatFormatting.RED, val));
+			return 0;
+		}
+
+		BigInteger newEMC = provider.getEmc();
+		switch(action) {
+			case ADD -> {
+				newEMC = newEMC.add(value);
+				ctx.getSource().sendSuccess(PELang.COMMAND_EMC_ADD_SUCCESS.translate(formatEMC(value), player.getDisplayName(), formatEMC(newEMC)), true);
+			}
+			case REMOVE -> {
+				newEMC = newEMC.subtract(value);
+				if (newEMC.compareTo(BigInteger.ZERO) < 0) {
+					ctx.getSource().sendFailure(PELang.COMMAND_EMC_NEGATIVE.translateColored(ChatFormatting.RED, formatEMC(value), player.getDisplayName()));
+					return 0;
+				}
+				ctx.getSource().sendSuccess(PELang.COMMAND_EMC_REMOVE_SUCCESS.translate(formatEMC(value), player.getDisplayName(), formatEMC(newEMC)), true);
+			}
+			case SET -> {
+				newEMC = value;
+				ctx.getSource().sendSuccess(PELang.COMMAND_EMC_SET_SUCCESS.translate(player.getDisplayName(), formatEMC(value)), true);
+			}
+			case TEST -> {
+				if(newEMC.compareTo(value) >= 0) {
+					ctx.getSource().sendSuccess(PELang.COMMAND_EMC_TEST_SUCCESS.translateColored(ChatFormatting.GREEN, player.getDisplayName(), formatEMC(value)), true);
+					return 1;
+				} else {
```

Comment:
Unneeded else, can just be implicit due to the fact the if statement always returns.